dask/dataframe/tests/test_multi.py

```diff
@@ -262,6 +264,168 @@ def test_sequential_joins():
     assert_eq(multi_join_pd, multi_join_dd)
 
 
+def test_merge_asof_index():
+    A = pd.DataFrame({'left_val': list('abcd' * 3)}, index=[1,3,7,9,10,13,14,17,20,24,25,28])
+    a = dd.from_pandas(A, npartitions=4)
+    B = pd.DataFrame({'right_val': list('xyz' * 4)}, index=[1,2,3,6,7,10,12,14,16,19,23,26])
+    b = dd.from_pandas(B, npartitions=3)
+
+    C = pd.merge_asof(A, B, left_index=True, right_index=True)
+    c = merge_asof(a, b, left_index=True, right_index=True)
+
+    assert_eq(c, C)
+
+
+def test_merge_asof_on():
+    A = pd.DataFrame({'a': [1, 5, 10], 'left_val': ['a', 'b', 'c']})
+    a = dd.from_pandas(A, npartitions=2)
+    B = pd.DataFrame({'a': [1, 2, 3, 6, 7], 'right_val': [1, 2, 3, 6, 7]})
+    b = dd.from_pandas(B, npartitions=2)
+
+    C = pd.merge_asof(A, B, on='a').set_index('a')
+    c = merge_asof(a, b, on='a')
+    assert_eq(c, C)
+
+
+def test_merge_asof_on_no_exact_matches():
```

Comment:
This could be parametrized and shared with the previous test I think. Something like

````python
@pytest.mark.parametrize('allow_exact_matches', [True, False])
def test_merge_asof_on(allow_exact_matches):
    ...
```